### PR TITLE
feat(push): add push source list command

### PR DESCRIPTION
**Pull request by olamothe**
## Proposed changes

Add a command that list the available push source in your organization. Will be needed for upcoming command that allows to push local files from disk, since you need to specify which source to use.

## Testing

- [X] Unit Tests:
- [ ] Functionnal Tests:
- [X] Manual Tests:

-----
https://coveord.atlassian.net/browse/CDX-453

### CODE REVIEW

File: packages/cli/src/commands/source/push/list.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  public static flags = {
+    ...cli.table.flags(),
+  };
+
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    const {flags} = this.parse(SourcePushList);
+    const authenticatedClient = new AuthenticatedClient();
+    const org = (await authenticatedClient.cfg.get()).organization;
```
Comment (y-lakhdar) [inline packages/cli/src/commands/source/push/list.ts:27]: Is it a styling preference that the `org` variable is declared at the beginning even if it is only used in the `if` block?
Really not important

**Review by y-lakhdar — APPROVED** [review]

File: packages/cli/src/commands/source/push/list.ts (hunk 1 above)
Comment (olamothe) [reply packages/cli/src/commands/source/push/list.ts:27]: Yeah, styling preference I guess :D

File: packages/cli/src/commands/source/push/list.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      buildAnalyticsFailureHook(this, flags, err)
+    );
+    throw err;
+  }
+
+  private flattenSourceModels(sourceModels: SourceModel[]) {
+    return sourceModels.map((sourceModel) => ({
+      ...sourceModel,
+      status: sourceModel.information?.sourceStatus?.type,
+      numberOfDocuments: sourceModel.information?.numberOfDocuments,
```
Comment (louis-bompart) [inline packages/cli/src/commands/source/push/list.ts:77]: 🤔 how will `cli.table` digest those values if one of them is undefined or null

**Review by louis-bompart — APPROVED** [review]
lgtm

File: packages/cli/src/commands/source/push/list.ts (hunk 2 above)
Comment (olamothe) [reply packages/cli/src/commands/source/push/list.ts:77]: You will get an empty row in the table.

**Pull request merged**